subsys/net/lib/coap/coap.c

```diff
@@ -217,6 +217,14 @@ static int parse_option(const struct coap_packet *cpkt,
 	}
 
 	if (option) {
```

Comment:
how about one if statement here instead of two?

Reply:
We need 2 checks because because we're looking at 3 possible scenarios here:
- It's not an error condition if this function gets an option parameter == NULL.  So we look for this condition.  This happens when a user passes in an array of coap_options to parse_options() and we run out of indexes before the end of the option data.  [NOTE: This *could* generate a warning of some sort as well.]
- The function has a valid option pointer, but the length of the value field isn't large enough to hold the incoming data (This is the error I'm looking for -- and something the developer should be made aware of because it's fixable.)
- The function has a valid option pointer, and the length is ok to use it.